lib/cylc/get_cycle_summaries.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2015 NIWA
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Get task state summaries from the state file; used by gsummary and scan."""
+# TODO - why not get these from the suite daemon?
+
+import re
+import sys
+from subprocess import Popen, PIPE
+from operator import itemgetter
+from itertools import groupby
+
+import cylc.flags
+
+def get_task_cycle_statuses_updatetime(host, suite, owner=None):
```

Comment:
I took this out of gsummary.py to use it in cylc-scan as well.  However, now I'm getting the state totals from the "suite identity" (scan) interface in the daemon, and this can go back to gsummary (just for computing stop summaries)...